lib/src/warnings.dart

```diff
@@ -405,7 +405,7 @@ class PackageWarningOptions {
           .forEach((PackageWarning kind) => newOptions.ignore(kind));
     }
     if (allowErrorsInPackages != null &&
-        !allowWarningsInPackages.contains(packageMeta.name)) {
+        !allowErrorsInPackages.contains(packageMeta.name)) {
```

Comment:
Hmmm.  This might actually cause some pretty significant behavior changes since allowWarningsInPackages doesn't cover both.  But it does seem like the right thing to do.